sklearn/_isotonic.pyx

```diff
@@ -62,9 +62,9 @@ def _inplace_contiguous_isotonic_regression(floating[::1] y, floating[::1] w):
             i = k
 
 
-def _make_unique(cnp.ndarray[dtype=floating] X,
-                 cnp.ndarray[dtype=floating] y,
-                 cnp.ndarray[dtype=floating] sample_weights):
+def _make_unique(const floating[::1] X,
```

Comment:
`[::1]` makes a memoryview C-contiguous. Since this vector is 1D, this is not useful, nor implemented initially (we would have instead `cnp.ndarray[dtype=floating, mode="c"]`). See this [useful doc](https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html).

Therefore you should prefer `const floating[:] X` instead, for all inputs `X`, `y` and `sample_weights`